inline-caching/README.md

```diff
@@ -0,0 +1,264 @@
+we have the following line
+
+```plain
+spring.data.gemfire.pool.locators=localhost[10334]
+```
+This tells the GemFire client the location of our GemFire cluster that we created above. 
+The port `10334` is the default port, but any port can be used if configured to do so.
+
+Starting the Spring Boot app with GemFire can be done from the inline-caching project root directory with
+
+```bash
+./gradlew bootRun
+```
+
+When the application starts, you should see in the console output that the GemFire client has discovered a locator.
+
+```plain
+AutoConnectionSource discovered new locators [...:10334]
+```
+
+This means that our Spring Boot web application is connected and ready to handle requests. 
+
+## Performing Requests
+
+**The following web requests can be made to test the Cache Loader**
+
+Add a value to Postgres database. This will bypass GemFire but will change a value in Postgres.
+
+    psql postgres -U myuser;
+    update pgbench_tellers set filler = 'hello' where tid = 1;
+
+Retrieve a value from the webservice. This will invoke GemFire's CacheLoader. GemFire shouldn't know of this 
+data with the cooresponding key `1` and therefore will reach out to Postgres.
+
+    curl localhost:8080/1          # should see the value hello
+
+Change a value on the Postgres database. Again, this will bypass GemFire but will update the value in the backing database. 
+This will setup a demonstration that GemFire is indeed caching values. 
+
+    psql postgres -U myuser;
+    update pgbench_tellers set filler = 'goodbye' where tid = 1;
+
+Retrieve a value from webservice. This will still show the originally set value because the web service / GemFire was not involved. 
+
+    curl localhost:8080/1          # should still see the value hello in the console because it is cached
+
+Clear the cache in GemFire using gfsh
+
+    remove --key=1 --region=item
+
+Retrieve the value from the webservice after clearing the GemFire cache so that the CacheLoader will be invoked again. 
+
+    curl localhost:8080/1          # should see the new value goodbye
+
+**The following web requests can be made to test the Async Event Listener**
+
+Write a value to the web service. This will write a value to GemFire first, and then the AsyncEventListener will pick up the write event and 
+persist the value to Postgres. 
+
+    curl -X PUT localhost:8080/1 -H 'Content-Type: application/json' -d '{"value":"potato"}'
+
+See the value in Postgres. We can see here that the value has been written by observing the return result in the console. 
+
+    psql postgres -U myuser;
+    select filler from pgbench_tellers where tid=1;  -- should see potato
+
+Retrieve the value from the web service. Just to be certain, we can see the new value has been updated for reads as well. 
+The AsyncEventListener and CacheLoader work together. 
+
+    curl localhost:8080/1          # should see the value potato
+
+
+## Cleanup
+
+#### Gemfire and Web applications
```

Comment:
This is not the way users should learn to stop GemFire.

They should be stopping it using GFSH.

Reply:
Especially if users have other Java things running, you don't want them stopping all Java processes.